src/document_basic.c

```diff
@@ -191,10 +194,30 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
 
     // on crdt the return value might be the underline value, we must copy it!!!
     // TODO: change `fs->text` to support hash or json not RedisModuleString
-    if (japi->getJSON(json, ctx, &doc->fields[oix].text) != REDISMODULE_OK) {
-      RedisModule_Log(ctx, "verbose", "Failed to load value from field %s", field->path);
-      goto done;
-    }
+
+    if (type != JSONType_Array) {
+      if (japi->getJSON(json, ctx, &doc->fields[oix].text) != REDISMODULE_OK) {
+        RedisModule_Log(ctx, "verbose", "Failed to load value from field %s", field->path);
+        goto done;
+      }
+    } else if (field->types & INDEXFLD_T_TAG) {
```

Comment:
```suggestion
    } else {
```

Reply:
For Array type, `INDEXFLD_T_TAG` was already tested in line 185
`(type == JSONType_Array && !(field->types & INDEXFLD_T_TAG))`
And we know it is enabled if reached here.